staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -562,6 +563,119 @@ func (s *store) List(ctx context.Context, key string, opts storage.ListOptions,
 	return s.list(ctx, key, opts, listObj, true)
 }
 
+type clientPagingConfig struct {
+	// hasMore guards if we need to set up the remaining configuration
+	hasMore         bool
+	startKey        string
+	resourceVersion int64
+	remainingCnt    int64
+}
+
+func (s *store) paginatedList(ctx context.Context, key, end string, rev int64, fromRev *uint64,
+	v reflect.Value, typeName string, pred storage.SelectionPredicate, newItemFunc func() runtime.Object) (pcfg *clientPagingConfig, err error) {
+	pcfg = &clientPagingConfig{}
+	maximumLimit := int64(500)
```

Comment:
It would be great if `500` was a flag.  Or at least a `const` with a descriptive name, so we know where to look for future performance tuning.

Reply:
yeah, it will be the `kube-apiserver` flag in the next commit.